lib/model/query/forms.js

```diff
@@ -761,6 +762,16 @@ const checkDeletedForms = (xmlFormId, projectId) => ({ maybeOne, context }) => (
     }
   }));
 
+const checkDatasetWarnings = (dataset) => ({ context }) => {
+  if (context.query.ignoreWarnings) return resolve();
+
+  if (dataset.isDefined() && dataset.get().warnings != null) {
+    if (!context.transitoryData.has('workflowWarnings')) context.transitoryData.set('workflowWarnings', []);
+    context.transitoryData.get('workflowWarnings').push(dataset.get().warnings);
```

Comment:
If you make `warnings` an array:

```suggestion
    context.transitoryData.get('workflowWarnings').push(...dataset.get().warnings);
```